clients/src/test/java/org/apache/kafka/clients/consumer/internals/LegacyKafkaConsumerTest.java

```diff
@@ -164,7 +163,7 @@
  * Note to future authors in this class. If you close the consumer, close with DURATION.ZERO to reduce the duration of
  * the test.
  */
-public class KafkaConsumerTest {
+public class LegacyKafkaConsumerTest {
```

Comment:
For my understanding, do we have similar tests for the new consumer? Or do we need to run a subset of those for it?

Reply:
We have similar tests. My other attempt at the delegate work was much _deeper_ so that we could reuse this test and run it against both implementations, but it got out of hand.

Reply:
Are you referring to `AsyncKafkaConsumerTest`? I wonder if we should try a bit more to keep this one for both implementation. In this PR, we could perhaps keep it as `KafkaConsumerTest` and configure it to only run against the legacy consumer and we could parameterised it further as a follow-up. Without `KafkaConsumerTest`, we don't have any other tests validating the main "interface" and there is a ton of tests in this one.

Reply:
I have updated all of the tests to support parameterization. However, about half of them only support the legacy protocol because when we exercise the new protocol, the test fails. There is an explanation in the Javadoc of the method that is used as a 'source' for the parameters that elaborates.